systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java

```diff
@@ -0,0 +1,334 @@
+                                new NetworkPolicyPeerBuilder()
+                                    .withNewPodSelector()
+                                    .withMatchLabels(matchLabelsForTls)
+                                    .endPodSelector()
+                                    .build())
+                        .endGenericKafkaListener()
+                    .endListeners()
+                .endKafka()
+            .endSpec()
+            .done();
+
+        String topic0 = "topic-example-0";
+        String topic1 = "topic-example-1";
+        KafkaTopicResource.topic(CLUSTER_NAME, topic0).done();
+        KafkaTopicResource.topic(CLUSTER_NAME, topic1).done();
+
+        String userName = "user-example";
+        KafkaUser kafkaUser = KafkaUserResource.scramShaUser(CLUSTER_NAME, userName).done();
+
+        KafkaClientsResource.deployKafkaClients(true, allowedKafkaClientsName, kafkaUser).done();
+
+        String allowedKafkaClientsPodName = kubeClient().listPodsByPrefixInName(allowedKafkaClientsName).get(0).getMetadata().getName();
+
+        LOGGER.info("Verifying that {} pod is able to exchange messages", allowedKafkaClientsPodName);
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(allowedKafkaClientsPodName)
+            .withTopicName(topic0)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .withKafkaUsername(userName)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .build();
+
+        internalKafkaClient.checkProducedAndConsumedMessages(
+            internalKafkaClient.sendMessagesTls(),
+            internalKafkaClient.receiveMessagesTls()
+        );
+
+        KafkaClientsResource.deployKafkaClients(true, deniedKafkaClientsName, kafkaUser).done();
+
+        String deniedKafkaClientsPodName = kubeClient().listPodsByPrefixInName(deniedKafkaClientsName).get(0).getMetadata().getName();
+
+        InternalKafkaClient newInternalKafkaClient = internalKafkaClient.toBuilder()
+            .withUsingPodName(deniedKafkaClientsPodName)
+            .withTopicName(topic1)
+            .withConsumerGroupName(ClientUtils.generateRandomConsumerGroup())
+            .build();
+
+        LOGGER.info("Verifying that {} pod is  not able to exchange messages", deniedKafkaClientsPodName);
+
+        assertThrows(AssertionError.class, () -> {
+            newInternalKafkaClient.checkProducedAndConsumedMessages(
+                newInternalKafkaClient.sendMessagesTls(),
+                newInternalKafkaClient.receiveMessagesTls()
+            );
+        });
+    }
+
+    @Test
+    void testNPWhenOperatorIsInSameNamespaceAsOperand() {
+        EnvVar operatorNamespaceEnv = new EnvVarBuilder()
+            .withName("STRIMZI_OPERATOR_NAMESPACE")
+            .withValue(NAMESPACE)
+            .build();
```

Comment:
It is configured in the installation files (https://github.com/strimzi/strimzi-kafka-operator/blob/e1e1035babc697e2b6fae34554c4503823ef5531/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml#L87) ... why do you need to configure it again your self here? Isn't that unnecessary? I think you deploy using these files ...

Reply:
Yep, good point, thanks